packages/utils/src/schema/sanitizeDataForNewSchema.ts

```diff
@@ -193,5 +196,12 @@ export default function sanitizeDataForNewSchema<
     }
     // Also probably want to deal with `prefixItems` as tuples with the latest 2020 draft
   }
+  // Schema contains a single scalar value
+  else {
```

Comment:
I'm not sure what are the other possible cases here. 

Logic before this branch:
- We checked that the schema  doesn't have a `properties` key
- This check is false old `schema type == array AND new schema type == array AND current data == array`

I'm only checking that newSchema has a `default` key but maybe there is a better check here?